### PR TITLE
Generate proof data

### LINKED ISSUE #718: Generate extensive data set using MPP fuzzer for unit test
Using the MPP fuzzer generate the extensive dataset which covers many cases to unit test Merkel Patricia Proof library. 

Sample cases: 
1. Full-length proof with all branch nodes. 
2. Proof with one extension node and all branch nodes. 
3. Proof with more than one extension node. (e,b,e,b,l) 
 
Write a script: 
 1.  which can be executed to generate test dataset using MPP fuzzer. 
 2. Which can write generated test dataset into a json file.

### PR DESCRIPTION
**Pull request by schemar**
Fixes #718

### CODE REVIEW

**Comment by schemar** [comment]
Ready for review

File: tools/fuzzy_proof_generator_tool/patterns.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const patterns: string[] = [
```
Comment (benjaminbollen) [inline tools/fuzzy_proof_generator_tool/patterns.ts]: should some patterns end on a branch node too?

Comment (schemar) [reply tools/fuzzy_proof_generator_tool/patterns.ts]: I will add those. Please node though, that due to the solidity implementation of the MPP library, we can only end on *two* branch nodes.

Working examples:
* `bbb`
* `bebb`

Non-working examples:
* `eb`
* `bbeb`

**Comment by schemar** [comment]
Moved to bounce, will add more patterns.

**Comment by schemar** [comment]
Ready for review again.

**Comment by schemar** [comment]
> Nice work 👍
> Overall looks good 🐼
> Question: Why values are stored as a buffer, not as a string like other files in `test/data` folder.
> Did you try to recreate the js buffer object from serialized data in `generatedProofData.json`?

@sarvesh-ost yes I convert them there:

https://github.com/OpenST/mosaic-contracts/pull/751/files#diff-45732ae0a73ed327c02d5c209394c4c8R27

Should I store them right away in hex instead?

**Comment by sarvesh-ost** [comment]
> > Nice work 👍
> > Overall looks good 🐼
> > Question: Why values are stored as a buffer, not as a string like other files in `test/data` folder.
> > Did you try to recreate the js buffer object from serialized data in `generatedProofData.json`?
> 
> @sarvesh-ost yes I convert them there:
> 
> https://github.com/OpenST/mosaic-contracts/pull/751/files#diff-45732ae0a73ed327c02d5c209394c4c8R27
> 
> Should I store them right away in hex instead?

I think yes, will reduce the overhead of reconverting to hex and this also makes it consistent to other JSON files in the same folder.

**Comment by schemar** [comment]
> > > Nice work 👍
> > > Overall looks good 🐼
> > > Question: Why values are stored as a buffer, not as a string like other files in `test/data` folder.
> > > Did you try to recreate the js buffer object from serialized data in `generatedProofData.json`?
> > 
> > 
> > @sarvesh-ost yes I convert them there:
> > https://github.com/OpenST/mosaic-contracts/pull/751/files#diff-45732ae0a73ed327c02d5c209394c4c8R27
> > Should I store them right away in hex instead?
> 
> I think yes, will reduce the overhead of reconverting to hex and this also makes it consistent to other JSON files in the same folder.

:+1: on it!

**Comment by schemar** [comment]
Ready for review again.
/cc @sarvesh-ost

**Pull request merged**